### ISSUE TITLE
Auto detect and set text direction (RTL/LTR)

### ISSUE THREAD

**Issue by mr-islam** [gsantner/markor#1221]
Hi, I really love Markor and had one feature request:

Auto detection of RTL text and setting RTL direction. (At least in view mode.)

Right now, there's a setting that changes text direction globally for all documents. That's already helpful and better than so many apps that don't event consider RTL.

But the issue is when a user has documents in multiple languages, some RTL and some LTR. In this case, a global setting is unsuitable.

If possible, I would like to request auto detection of RTL text to set the direction of each document. 

Another, possibly more difficult, option would be auto detection of text direction on the _paragraph_ level. This would help when one file has multiple languages.

An easy way to do this is to check the text-direction unicode property of the first character in each paragraph and set the paragraph direction accordingly.

I'm not too familiar with Android app development, so sadly I don't think I can contribute to this directly with code...

**Comment by gsantner** [comment]
I read through some discussions and topics - everybody/everything I read does RTL/LTR based on Local and/or system settings. Not based on any content.  
Analyzing content comes at cost at performance and I don't see the need for it in this case.

What I rather think off is, moving the RTL setting out from global settings to file settings menu so you can manually set it per-file as you wish.

**Comment by mr-islam** [comment]
Yes of course, doing it on a per file basis is a popular approach that users are familiar with 👍. 

I was suggesting automatic text direction based on Google Docs and Keep – they have my personal favorite implementation with auto direction based on the content of each paragraph. 

But that has performance costs like you said, and is not very common.

**Comment by ashraf2033** [comment]
I'm working on this currently but I'm having hard reflecting the setting on the UI, I've already  stored the app settings values but I'm not able to find a way to  load the note with the rtl.

most existing file settings are related to hleditor, which is not the case for the Text direction in View mode 

I've added the following lines in AppSettings.java
```

  private static final String PREF_PREFIX_RTL_STATE = "PREF_PREFIX_RTL_STATE";


 public void setDocumentRtlState(final String path, final boolean state) {
        final boolean _default = isRenderRtl();
        if (fexists(path)) {
            setBool(PREF_PREFIX_RTL_STATE + path, state);
        }
    }

    public boolean getDocumentRtlState(final String path) {
        final boolean _default = isRenderRtl();
        if (!fexists(path)) {
            return _default;
        } else {
            return getBool(PREF_PREFIX_RTL_STATE + path, _default);
        }
    }

```


and these are my additions to DocumentEditAndViewFragment
```

   case R.id.action_enable_rtl: {
                final boolean newState = !_appSettings.isRenderRtl();

                      _appSettings.setDocumentRtlState(_document.path, newState);
                        updateMenuToggleStates(0);

                    }

                return true;


    if ((mi = _fragmentMenu.findItem(R.id.action_enable_rtl))!= null){
            mi.setChecked(_appSettings.getDocumentRtlState(_document.path));

```
